src/Array.mo

```diff
@@ -581,7 +583,7 @@ module {
   /// Space: O(1)
   ///
   /// *Runtime and space assumes that `combine` runs in O(1) time and space.
-  public func foldRight<X, Y>(array : [X], base : Y, combine : (X, Y) -> Y) : Y {
+  public func foldRight<X, A>(array : [X], base : A, combine : (X, A) -> A) : A {
```

Comment:
nit: we call the type parameters `X, Y` for some functions and `X, A` for other functions. What's the reasoning behind the naming?

Reply:
I've been using `X` for type parameters, and `Y` if there is a second generic type.
Only exception is if there is some meaning to the type beyond "a second generic type".

In this case, `A` is the accumulator type. Or in `mapResult`, `E` is the error type.

But not sure if this is actually helpful or consistent